### PR TITLE
Add plugin: Encoder/Decoder

### PR DESCRIPTION
**Pull request by rudimuc**
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: https://github.com/rudimuc/obsidian-coder

## Release Checklist
- [X] I have tested the plugin on
  - [X]  Windows
  - [ ]  macOS
  - [X]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [X] My GitHub release contains all required files
  - [X] `main.js`
  - [X] `manifest.json`
  - [X] `styles.css` _(optional)_
- [X] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [X] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [X] My README.md describes the plugin's purpose and provides clear usage instructions.
- [X] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [X] I have added a license in the LICENSE file.
- [X] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

### CODE REVIEW

**Comment by joethei** [comment]
["description": "This plugin converts texts into base64 and other formats.",](https://github.com/rudimuc/obsidian-coder/blob/8afa47eecc8b315a0fb5d024c8750ed540c4febc/manifest.json#L6)
Avoid including sentences like "This plugin does ..." in your description.
We have published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.
Also, the part "and other formats" is not true at the moment.

[// Remember to rename these classes and interfaces!](https://github.com/rudimuc/obsidian-coder/blob/8afa47eecc8b315a0fb5d024c8750ed540c4febc/main.ts#L5)
Please do this.

[[1](https://github.com/rudimuc/obsidian-coder/blob/8afa47eecc8b315a0fb5d024c8750ed540c4febc/main.ts#L86)][[2](https://github.com/rudimuc/obsidian-coder/blob/8afa47eecc8b315a0fb5d024c8750ed540c4febc/main.ts#L88)][[3](https://github.com/rudimuc/obsidian-coder/blob/8afa47eecc8b315a0fb5d024c8750ed540c4febc/main.ts#L105)][[4](https://github.com/rudimuc/obsidian-coder/blob/8afa47eecc8b315a0fb5d024c8750ed540c4febc/main.ts#L107)]:[Avoid using `innerHTML`](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%60innerHTML%60%2C+%60outerHTML%60+and+%60insertAdjacentHTML%60)

[interface MyPluginSettings {](https://github.com/rudimuc/obsidian-coder/blob/8afa47eecc8b315a0fb5d024c8750ed540c4febc/main.ts#L7-L13), [async loadSettings() {](https://github.com/rudimuc/obsidian-coder/blob/8afa47eecc8b315a0fb5d024c8750ed540c4febc/main.ts#L67-L73)
Since your plugin does not use settings, please remove all references to them.

**Comment by rudimuc** [comment]
Hi @joethei ,
thank you for your input. I have changed the code to be compliant with your rules.
Please have a look.

**Comment by liamcain** [comment]
- [var](https://github.com/rudimuc/obsidian-coder/blob/c229c3e4f28ba8fa0db4cdf08960258599a59fa5/main.ts#L57) avoid using`var` since it has function-level scope, so it can easily lead to bugs if you're not careful. Here's a pretty good guide with examples of why not to use `var`: [why `var` is obsolete](https://javascript.plainenglish.io/4-reasons-why-var-is-considered-obsolete-in-modern-javascript-a30296b5f08f)

File: community-plugins.json [hunk 1]
```diff
@@ -9504,5 +9504,12 @@
     "description": "Multi-regex management for data linking and batch processing across selection, files, and folders.",
     "author": "Syr",
     "repo": "Syr0/R.E.L.A.X."
+  },
+  {
+    "id": "coder",
+    "name": "Encoder",
```
Comment (liamcain) [inline community-plugins.json:9510]:
```suggestion
    "name": "Encoder/Decoder",
```

**Pull request merged**